### DISCUSSION TITLE
Addons needing maintainers

### DISCUSSION THREAD

**Discussion by kaosine** [pulsar-edit/.github#35] [Announcements]
As we need a terminal panel, add-on or something else eventually I'm linking these as possible alternatives and a copy of this is over on discord:

- https://github.com/vertexclique/tokamak-terminal
- https://github.com/jeremyramin/terminal-plus
- https://github.com/briederer/atom-terminal-tab

Just not sure which someone would eventually want to maintain but all of them from my memory contain a MIT license so these should be fair game to fork and implement somehow as add-ons or an internal project eventually once things calm down

Others worth mentioning and in need of a maintainer for usefulness (anyone can suggest more, and I'll add them over time to this list or remove as maintainers are found):

- https://github.com/ArnaudRinquin/atom-zentabs  - this is a productivity add-on and a fairly useful one when it worked. This I would love to have someone internalize eventually as it's a useful feature for a person with adhd/asd like me. It keeps your tabs down to a minimum by closing older ones as you open newer or prevents opening newer(I forget, but all the same tl;dr is it's a tab management that limits tab amounts).

As well until further notice please refer to this for packages that we have backed up. It will need to be sorted and put use but until this is up in client and on a website with proper search please ignore the mess as this was a grabbag to make sure we had all these backed up. Attempting to contact the dev is recommended but knowing how long it's been since updates It's doubtful that they'll respond. But this will tell you the file you can find the relevant info in for the add-on:
https://github.com/confused-Techie/atom-package-collection/blob/main/out/package_pointer.json

Also a link to our discord if you happened to have missed it since a fair chunk of our discourse takes place over there:

https://discord.gg/7aEbB9dGRT

**Comment by sertonix** [comment]
Another package to maintain would also be [ftp-remote-edit](https://github.com/h3imdall/ftp-remote-edit). Remote editing is very useful. It would also be possible to provide an alternative with the same or more functions.

Other packages searching for maintainer:
- [`ruby-solargraph`](https://github.com/castwide/atom-solargraph/issues/23)
- [`php-ide-serenata`](https://github.com/Gert-dev/php-ide-serenata/issues/522)

**Comment by sertonix** [comment]
There are a few packages that were forked by Atom but not maintained.
So ether they are maintained or the repositories should be removed/archived.

 * [atom-ternjs](https://github.com/pulsar-edit/atom-ternjs)
 * [atom-bugs](https://github.com/pulsar-edit/atom-bugs)
 * [tree-view-finder](https://github.com/pulsar-edit/tree-view-finder)
 * [atom-trello](https://github.com/pulsar-edit/atom-trello)
 * [scroll-searcher](https://github.com/pulsar-edit/scroll-searcher)
 * [tree-view-search-bar](https://github.com/pulsar-edit/tree-view-search-bar)
 * [project-ring](https://github.com/pulsar-edit/project-ring)
 * [editor-background](https://github.com/pulsar-edit/editor-background)

**Reply by sertonix** [reply]
Is there any way we move the upstream from these repositories to the main repositories? That would make the situation more clear.

**Reply by sertonix** [reply]
> Is there any way we move the upstream from these repositories to the main repositories? That would make the situation more clear.

Ok, answering my own question. There is no way to do that other than maybe asking the github support. So the easiest way would be to delete the repositories and fork them directly from the original repository.

**Comment by sertonix** [comment]
There are also packages created by Atom that are unmaintained:
* [notebook](https://github.com/pulsar-edit/notebook)
* [town-crier](https://github.com/pulsar-edit/town-crier)
* [revert-buffer](https://github.com/pulsar-edit/revert-buffer)

**Comment by Daeraxa** [comment]
As per [Discord discussion](https://discord.com/channels/992103415163396136/992103415163396139/1060402448986013706) from UziTech:

https://github.com/semantic-release/apm
https://github.com/semantic-release/apm-config

**Comment by sertonix** [comment]
The package [mcfunction-novum](https://github.com/jan00bl/mcfunction-novum) is also searching for maintainers.
You can contact me or the owner if you have questions about it.
It would need a bit of rebranding from Atom to Pulsar.
The plan for the future was to implement the [Spyglass language server](https://github.com/SPYGlassMC/SPYGlass). Maybe with [generic-lsp](https://github.com/mauricioszabo/generic-lsp).
Note that the Spyglass community is not a fan of Atom.

**Comment by savetheclocktower** [comment]
When I am done with my current windmill-tilt of modernizing the tree-sitter integration, I could see myself maintaining a fork of [atom-zentabs](https://github.com/ArnaudRinquin/atom-zentabs), along with a few other extensions — ideally whichever ones seem like they won't be long-term maintenance headaches.

**Comment by confused-Techie** [comment]
I want to go ahead and drop this link here.

As many already know due to license issues we are having to remove several packages from the registry. This means the concept of those packages is now unfulfilled, and especially if anyone knows the author or is otherwise able to contact them we could look at including those packages again. But take a look at the [Backend Admin Actions](https://github.com/pulsar-edit/package-backend/blob/main/docs/reference/Admin_Actions.md) log for these packages that must be removed, and see if there's something vital we are now missing. (This document will be continually updated, so you can always check back in again)
